share/spack/templates/container/singularity.def

```diff
@@ -1,8 +1,17 @@
 Bootstrap: docker
-From: {{ build.image }}:{{ build.tag }}
+From: {{ build.image }}
 Stage: build
 
 %post
+{% if os_packages_build.list %}
+  # Update, install and cleanup of system packages needed at build-time
+{% if os_package_update %}
```

Comment:
> Does the template format require that there is no spaces before the directive?

The spaces before the directives are rendered, at least by default. Adding the spaces there would result in a less readable file once the template is instanced. I can check if `jinja2` has options to interpret a line with a directive as something that shouldn't be rendered.

Reply:
Adding two spaces in front of this `%if` is possible, but we have to carefully trim whitespaces [using + or - signs](https://jinja.palletsprojects.com/en/2.11.x/templates/#whitespace-control) at the end of the directive and the reader of the code has to understand that writing:
```
{% if ... %}
``` 
is different from
```
  {% if ... -%}
``` 
I would keep this modification as a follow up after v0.16.0, if you still think it's good to have. In any case it would be an internal refactor without changes to user facing behavior.

Reply:
> I would keep this modification as a follow up after v0.16.0, if you still think it's good to have

I think this is important to have. I'd like to figure out how to address the template whitespace issues before merging this PR.

I think you are saying that to use proper indentation in the template, you will have to add this formatting syntax so that the resulting generated file does not contain the extra whitespace. To clarify my original suggestion in https://github.com/spack/spack/pull/15028#discussion_r451228201, I was saying that for now just indent the directives in the template file and allow the resulting generated files to contain additional whitespace.

I agree that the manual whitespace management syntax looks unpleasant and not worth adding. The documentation you linked also appears to contain a universal setting for removing whitespace associated with template directives (e.g. constructing the `jinja2.Environment` with `trim_blocks=True` and `lstrip_blocks=True`) - is there a reason not to do that?

Reply:
To be clear, one of my proposed solutions in https://github.com/spack/spack/pull/15028#discussion_r524813712 is hopefully very easy (but results in messier generated files), so if that is acceptable then I think this should be quick to resolve.